easybuild/easyblocks/b/bwise.py

```diff
@@ -0,0 +1,120 @@
+from easybuild.tools.filetools import copy_file, which, write_file
+from easybuild.tools.modules import get_software_root
+
+
+class EB_BWISE(MakeCp):
+    """
+    Custom easyblock to install BWISE
+    """
+    @staticmethod
+    def extra_options():
+        """Change default values of options"""
+        extra = MakeCp.extra_options()
+        # files_to_copy is not mandatory here
+        extra['files_to_copy'][2] = CUSTOM
+        return extra
+
+    def build_step(self):
+        """Run multiple times for different sources"""
+
+        # BCALM is a git submodule of BWISE but we use it as a dependency
+        bcalm = get_software_root('BCALM')
+        if not bcalm:
+            raise EasyBuildError("BWISE needs BCALM to work")
+
+        makefiles_fixes = [
+            (r'^CC=.*$', 'CC=$(CXX)'),
+            (r'^CFLAGS=.*$', 'CFLAGS:=$(CFLAGS)'),
+            (r'^LDFLAGS=.*$', 'LDFLAGS:=$(LDFLAGS) -fopenmp')
+        ]
+
+        def find_subdir(pattern):
+            """Changes to the sub directory that matches the given pattern"""
+            subdir = glob.glob(os.path.join(self.builddir, pattern))
+            if subdir:
+                return subdir[0]
+            else:
+                raise EasyBuildError("Could not find a subdirectory matching the pattern %s", pattern)
+
+        # BWISE has 3 independant parts, we build them one by one
+        # first BWISE itself
+        subdir = find_subdir(os.path.join('BWISE-*', 'src'))
+        change_dir(subdir)
+        apply_regex_substitutions('makefile', makefiles_fixes)
+        apply_regex_substitutions(os.path.join(subdir, '..', 'Bwise.py'),
+                                  [(r'^BWISE_MAIN = .*$', 'BWISE_MAIN = os.environ[\'EBROOTBWISE\']')])
+        super(EB_BWISE, self).build_step()
+
+        # Onwards to BGREAT
+        subdir = find_subdir('BGREAT2-*')
+        change_dir(subdir)
+        apply_regex_substitutions('makefile', makefiles_fixes)
+        super(EB_BWISE, self).build_step()
+        copy_file(os.path.join(subdir, 'bgreat'), self.cfg['start_dir'])
+
+        # Finally, BTRIM
+        subdir = find_subdir('BTRIM-*')
+        change_dir(subdir)
+        apply_regex_substitutions('makefile', makefiles_fixes)
+        super(EB_BWISE, self).build_step()
+        copy_file(os.path.join(subdir, 'btrim'), self.cfg['start_dir'])
+
+        binaries = ['sequencesToNumbers', 'numbersFilter', 'path_counter', 'maximal_sr', 'simulator',
+                    'path_to_kmer', 'K2000/*.py', 'K2000/*.sh']
+        self.cfg['files_to_copy'] = [(['bgreat', 'btrim', 'Bwise.py'] + ['src/%s' % x for x in binaries], 'bin'),
+                                     'data']
+
+    def install_step(self):
+        super(EB_BWISE, self).install_step()
+
+        # BWISE expects BCALM to be at exactly this location...
+        bcalm = which('bcalm')
+        txt = """#!/bin/sh
+%s "$@"
+        """ % bcalm
```

Comment:
I don't think injecting the full path to `bcalm` is a good idea... Just leave this wrapper as it was?